homeassistant/components/epson/media_player.py

```diff
@@ -127,6 +127,17 @@ async def async_update(self):
         else:
             self._state = STATE_OFF
 
+    @property
+    def device_info(self):
+        """Get attributes about the device."""
+        return {
+            "identifiers": {(DOMAIN, self._unique_id)},
```

Comment:
Only return device info if we have a unique_id.

Reply:
through config flow we have unique id (serial number) of projector, is this enough?

Reply:
Yes, but the old config entries may not have a unique id yet. So we need to guard for this somehow. Preferably we shouldn't add the entity at all unless we have all data. Instead get the serial if it's missing and update the config entry and then reload the integration as in my other comment.

Reply:
Alternatively we could add the entity as unavailable if the device is off until we can get the serial and then update and reload. But then we need to not return incorrect device info here.